guiEditor/src/components/parentingPropertyGridComponent.tsx

```diff
@@ -1,75 +1,50 @@
 import * as React from "react";
-import { OptionsLineComponent } from "../sharedUiComponents/lines/optionsLineComponent";
-import { LineContainerComponent } from "../sharedUiComponents/lines/lineContainerComponent";
-import { GlobalState } from "../globalState";
 import { Control } from "babylonjs-gui/2D/controls/control";
-import { Container } from "babylonjs-gui/2D/controls/container";
+import { TextLineComponent } from "../sharedUiComponents/lines/textLineComponent";
+import { FloatLineComponent } from "../sharedUiComponents/lines/floatLineComponent";
+import { LockObject } from "../sharedUiComponents/tabs/propertyGrids/lockObject";
+import { PropertyChangedEvent } from "../sharedUiComponents/propertyChangedEvent";
+import { Observable } from "babylonjs/Misc/observable";
+import { Grid } from "babylonjs-gui/2D/controls/grid";
 
 interface IParentingPropertyGridComponentProps {
-    guiNode: Control;
-    guiNodes: Control[];
-    globalState: GlobalState;
+    control: Control,
+    lockObject: LockObject,
+    onPropertyChangedObservable?: Observable<PropertyChangedEvent>
 }
 
 export class ParentingPropertyGridComponent extends React.Component<IParentingPropertyGridComponentProps> {
     constructor(props: IParentingPropertyGridComponentProps) {
         super(props);
+        const cellInfo = (this.props.control.parent as Grid).getChildCellInfo(this.props.control);
+        this._rowNumber = parseInt(cellInfo.substring(0, cellInfo.search(":")));
+        this._columnNumber = parseInt(cellInfo.substring(cellInfo.search(":") + 1));
     }
-    public parentIndex: number = 0;
-
-    public addChildGui(childNode: Control, parentNode: Control) {
-        this.props.globalState.guiTexture.removeControl(childNode);
-        (parentNode as Container).addControl(childNode);
-    }
-
-    public removeChildGui(childNode: Control, parentNode: Control) {
-        (parentNode as Container).removeControl(childNode);
-        this.props.globalState.guiTexture.addControl(childNode);
+    _columnNumber: number;
+    _rowNumber: number;
+
+    updateGridPosition()
+    {
+        const grid = this.props.control.parent as Grid;
+        this.props.control.parent?.removeControl(this.props.control);

```

Comment:
this is the reference to grid, so if it doesn't exist it should fail. Why not use grid here?